pennylane/transforms/qfunc_transforms.py

```diff
@@ -0,0 +1,386 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contains tools and decorators for registering qfunc transforms."""
+# pylint: disable=too-few-public-methods
+import functools
+import inspect
+
+import pennylane as qml
+
+
+def make_tape(fn):

```

Comment:
In the last couple of weeks, I've actually noticed a _lot_ of places in the core codebase where having access to `make_tape` would be super useful. For example, there are a lot of places currently doing:

```python
active_tape = get_active_tape()

if active_tape is None:
    with QuantumTape() as tape:
        fn(*args, **kwargs)
else:
    with active_tape.stop_recording(), QuantumTape() as tape:
        fn(*args, **kwargs)

# do something with tape
```

So it would be a huge benefit to maintainability and bug squashing to simply replace these occurrences with
```python
tape = make_tape(fn)(*args, **kwargs)
```

Hence why I didn't make this private 😆